### PR TITLE
Docs for changes related to HonorPVReclaimPolicy feature

### PR DESCRIPTION
**Pull request by deepakkinni**
Doc for KEP https://github.com/kubernetes/enhancements/issues/2644

Details present at https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2644-honor-pv-reclaim-policy

Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

**Comment by deepakkinni** [comment]
cc: @sftim

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 1]
```diff
@@ -175,6 +175,64 @@ spec:
 
 However, the particular path specified in the custom recycler Pod template in the `volumes` part is replaced with the particular path of the volume that is being recycled.
 
+### PersistentVolume Deletion Protection Finalizer
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:178]:
```suggestion
### PersistentVolume deletion protection finalizer
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 2]
```diff
@@ -175,6 +175,64 @@ spec:
...
+{{< feature-state for_k8s_version="v1.24" state="beta" >}}
+
+Finalizers are added on the PersistentVolume to ensure that the PersistentVolume having `Delete` reclaim policy is deleted only after the underlying storage assets are deleted.
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:181]:
```suggestion
Finalizers can be added on a PersistentVolume to ensure that PersistentVolumes
having `Delete` reclaim policy are deleted only after the backing storage are deleted.
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 3]
```diff
@@ -175,6 +175,64 @@ spec:
...
+Capacity:        1Gi
+Message:
+Source:
+    Type:          HostPath (bare host directory volume)
+    Path:          /tmp/data
+    HostPathType:
+Events:            <none>
+```
+
+The finalizer `external-provisioner.volume.kubernetes.io/finalizer` is added for csi volumes. The following is an example
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:205]:
```suggestion
The finalizer `external-provisioner.volume.kubernetes.io/finalizer` is added for CSI volumes.
The following is an example:
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 4]
```diff
@@ -175,6 +175,64 @@ spec:
...
+    Driver:            csi.vsphere.vmware.com
+    FSType:            ext4
+    VolumeHandle:      44830fa8-79b4-406b-8b58-621ba25353fd
+    ReadOnly:          false
+    VolumeAttributes:      storage.kubernetes.io/csiProvisionerIdentity=1648442357185-8081-csi.vsphere.vmware.com
+                           type=vSphere CNS Block Volume
+Events:                <none>
+```
+
+Enabling the `CSIMigration` feature for a specific in-tree plugin volume will remove
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:231]:
```suggestion
Enabling the `CSIMigration` feature for a specific in-tree volume plugin will remove
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 5]
```diff
@@ -175,6 +175,64 @@ spec:
...
+the `kubernetes.io/pv-controller` finalizer, while `external-provisioner.volume.kubernetes.io/finalizer`
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:232]:
```suggestion
the `kubernetes.io/pv-controller` finalizer, while adding the `external-provisioner.volume.kubernetes.io/finalizer`
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 6]
```diff
@@ -175,6 +175,64 @@ spec:
...
+is added. Similarly, disabling `CSIMigration` will remove the `external-provisioner.volume.kubernetes.io/finalizer`
+finalizer, while `kubernetes.io/pv-controller` finalizer is added.
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:234]:
```suggestion
finalizer, while adding the `kubernetes.io/pv-controller` finalizer.
```

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 7]
```diff
@@ -175,6 +175,66 @@ spec:
...
+    VolumeHandle:      44830fa8-79b4-406b-8b58-621ba25353fd
+    ReadOnly:          false
+    VolumeAttributes:      storage.kubernetes.io/csiProvisionerIdentity=1648442357185-8081-csi.vsphere.vmware.com
+                           type=vSphere CNS Block Volume
+Events:                <none>
+```
+
+Enabling the `CSIMigration` feature for a specific in-tree volume plugin will remove
+the `kubernetes.io/pv-controller` finalizer, while adding the `external-provisioner.volume.kubernetes.io/finalizer`
+is added. Similarly, disabling `CSIMigration` will remove the `external-provisioner.volume.kubernetes.io/finalizer`
```
Comment (tengqm) [inline content/en/docs/concepts/storage/persistent-volumes.md:235]:
```suggestion
finalizer. Similarly, disabling `CSIMigration` will remove the `external-provisioner.volume.kubernetes.io/finalizer`
```

**Comment by tengqm** [comment]
/approve cancel
This should target the 'dev-1.24' branch, rather than 'main'.

**Comment by deepakkinni** [comment]
> /approve cancel This should target the 'dev-1.24' branch, rather than 'main'.

I didn't get this, why should this change not be on main and only on dev-1.24?

**Comment by tengqm** [comment]
@deepakkinni The 'main' branch is a collection of docs for kubernetes 1.23. However, 'HonorPVReclaimPolicy' feature and the associated gates are all for 1.24. We the docs team caches all changes that happen to 1.24 in the 'dev-1.24' branch. When the 1.24 release is out of the door, we do a quick branch switch. The 'dev-1.24' becomes the live site, showing people documentation about 1.24. This is a little bit different from how code repos are using branches.

**Comment by deepakkinni** [comment]
> @deepakkinni The 'main' branch is a collection of docs for kubernetes 1.23. However, 'HonorPVReclaimPolicy' feature and the associated gates are all for 1.24. We the docs team caches all changes that happen to 1.24 in the 'dev-1.24' branch. When the 1.24 release is out of the door, we do a quick branch switch. The 'dev-1.24' becomes the live site, showing people documentation about 1.24. This is a little bit different from how code repos are using branches.

Thanks for the explanation. I've changed it to `dev-1.24`

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 8]
```diff
@@ -175,6 +175,66 @@ spec:
...
+having `Delete` reclaim policy are deleted only after the backing storage are deleted.
+
+The finalizer `kubernetes.io/pv-controller` is added to in-tree plugin volumes. The following is an example
+
+```shell
+kubectl describe pv task-pv-volume
+Name:            task-pv-volume
+Labels:          type=local
+Annotations:     <none>
+Finalizers:      [kubernetes.io/pv-protection, kubernetes.io/pv-controller]
```
Comment (xing-yang) [inline content/en/docs/concepts/storage/persistent-volumes.md:191]: `kubernetes.io/pv-controller` is added in 1.24 as beta.
`kubernetes.io/pv-protection` has been there for a while.
I wonder if this gives people impression that both are added in 1.24?  We probably should add a note to clarify.

**Pull request merged**